### PR TITLE
Add support for Hilla

### PR DESCRIPTION
**Pull request by platosha**
Hello! Let me kindly ask for adding the Hilla entry again, now that we have added support for Gradle in the Hilla 2.1 feature release.

This PR is a follow-up for #1113, which was reverted previously. Also see the related entry proposal: #872

### CODE REVIEW

File: start-client/dev/api.mock.json [hunk 1]
```diff
@@ -176,6 +176,18 @@
               },
               "reference": { "href": "https://vaadin.com/docs" }
             }
+          },
```
Comment (snicoll) [inline start-client/dev/api.mock.json:179]: This is a mock file for testing and doesn't need to be updated.

File: start-site/src/main/resources/application.yml [hunk 2]
```diff
@@ -163,6 +163,13 @@ initializr:
             version: 24.0.7
           - compatibilityRange: "[3.1.0-M1,3.2.0-M1)"
             version: 24.1.0
+      hilla:
```
Comment (snicoll) [inline start-site/src/main/resources/application.yml:166]: BOMs are ordered alphabetically.

**Review by snicoll — COMMENTED** [review]
Thanks for the PR. I would need to play with this a bit more but I've added a few comments for your consideration.

**Comment by platosha** [comment]
> Thanks for the PR. I would need to play with this a bit more but I've added a few comments for your consideration.

Thank you. Addressed the comments so far.

**Comment by snicoll** [comment]
@platosha can you please squash all that and rebase on top of `main` please?

**Comment by platosha** [comment]
@snicoll sure, done so

**Pull request closed without merging**